### ISSUE TITLE
'An internal IBM MQ error has occurred' when starting mqdevserver

### ISSUE THREAD

**Issue by meken** [ibm-messaging/mq-container#384]
I'm trying to run the container on [Azure Container Instances](https://azure.microsoft.com/en-us/services/container-instances/), but keep having the same failure every time I launch it. This could very well be related to the environment (Azure Container Instances), but before starting complaining to the Azure folks, it would be good to understand the root cause. The exit code 71 indicates insufficient resources, but I've verified (through mqconfig) that the instance is configured properly, and the same setup (with similar limits on CPU/MEM) works just fine on K8S (AKS).

I've attached the error logs (and some debug info), so any clarity would be appreciated.

Note, that I noticed there's another issue related to Azure with exactly the same error code (#327), however that seems to be caused by a volume mapping, which I'm not attempting. 

This may be a stretch, but anyone interested to reproduce this, you could do that using the following commands.
```bash
az group create -n rg-mq-test -l westeurope
az container create -g rg-mq-test -n aci-mqserver \
   --image ibmcom/mq --dns-name-label aci-mqserver --ports 1414 \
   --environment-variables LICENSE=accept MQ_QMGR_NAME=QM1 DEBUG=true \
   --cpu 4 --memory 4 --command-line "tail -f /dev/null"
az container exec -g rg-mq-test -n aci-mqserver --exec-command "/bin/bash"
bash-4.4$ runmqdevserver
```

Contents of ```/var/mqm/errors```: [errors.zip](https://github.com/ibm-messaging/mq-container/files/4096715/errors.zip)
Output of ```mqconfig```: [mqconfig.txt](https://github.com/ibm-messaging/mq-container/files/4096717/mqconfig.txt)
Output of ```runmqdevserver```: [stdout.txt](https://github.com/ibm-messaging/mq-container/files/4096718/stdout.txt)

**Comment by arthurbarr** [comment]
Thanks for all the debug output — that's really useful.  You're not using an external volume for the Queue Manager data, and I suspect that the filesystem used inside the container is not compatible with MQ.  I think something similar happens if you (say) use `tmpfs` as the filesystem — MQ tries to open a UNIX socket on the filesystem, but it can't do it due to filesystem limitations.

Can you try using a volume?

**Comment by meken** [comment]
Thanks for the insights, I've managed to run on a similar service on Azure without volumes, so I was hoping to get the same functionality with Container Instances. Nevertheless, I just tried it with a volume mapping without any luck. The file share (used for the volume) is mounted as root:root :( and that's causing problems as the mq server is run as user mqm and hence can't write to that folder. Apparently I'm not the only one [with that problem](https://github.com/MicrosoftDocs/azure-docs/issues/16481#issuecomment-496320087) though... 

BTW, I see that the data folder is created (so, at least the volume mapping is working), I'm assuming that this is done before the user is set to mqm.

I've attached again the output of runmqdevserver, but I don't think there's much that can be done about this specific error on your side as the volume mapping limitations are on Azure side.

[stdout-volume.txt](https://github.com/ibm-messaging/mq-container/files/4097409/stdout-volume.txt)

If you can think of anything that might cause the original issue, not related to volumes (as things seem to work on AKS, which is probably based on same implementation, without volumes), that would be appreciated.

**Comment by arthurbarr** [comment]
Based on further analysis from the core MQ development team, we think this is a host name length issue.  Socket addresses on Linux have a maximum length of 108 characters, and MQ includes the host name in the socket address.  The host name in this case is really long: "wk-caas-37a944df69294ee2b85acaec1b6e47b6-dee85ad9f498e5c0231a32".

You can set the `MQS_IPC_HOST` environment variable to change the name of the host from MQ's perspective, and get it to use a shorter name for the socket.

**Comment by meken** [comment]
Yes! That seems to fix the issue. Thanks a million for the swift response.
